### PR TITLE
Support of spinnerMode property to allow spinner datepicker on Android

### PR DESCRIPTION
**Pull request by SilVdB**
# Motivation

Since Android 5.0 (Material UI), the DatePicker component (https://developer.android.com/reference/android/widget/DatePicker.html) supports two modes:
- calendar
- spinner

We wanted to be able to use the spinner mode, since some of our Android users were complaining about the calendar mode being confusing and causing too much cognitive strain.
# Solution

I discovered that the easiest way to accomplish this is described here: http://stackoverflow.com/questions/27709668/hide-calenderview-from-datepicker-programatically-in-api-21/31610267#31610267

I added a property Boolean `spinnerMode` to the datepicker module (java) that decides wether or not the style suggested above is applied, resulting in a spinner in stead of a calendar view.

**Test plan (required)**
When opening a datepicker on Android, simply pass the property like this:

`DatePickerAndroid.open({
     date: Date.now(), 
     spinnerMode: true
`})

The resulting datepicker looks like this:
unspecified-3

When setting spinnerMode to false (or not passing it at all), the datepicker looks like this:
unspecified-2

### CODE REVIEW

File: ReactAndroid/src/main/java/com/facebook/react/modules/datepicker/DatePickerDialogFragment.java [hunk 1]
```diff
@@ -53,8 +53,13 @@ public Dialog onCreateDialog(Bundle savedInstanceState) {
     final int month = c.get(Calendar.MONTH);
     final int day = c.get(Calendar.DAY_OF_MONTH);
 
-    final DatePickerDialog dialog =
-        new DismissableDatePickerDialog(activityContext, onDateSetListener, year, month, day);
+    DatePickerDialog dialog;
+    if (args != null && args.containsKey(DatePickerDialogModule.ARG_SPINNERMODE) && args.getBoolean(DatePickerDialogModule.ARG_SPINNERMODE)) {
```
Comment (AaaChiuuu) [inline ReactAndroid/src/main/java/com/facebook/react/modules/datepicker/DatePickerDialogFragment.java]: do we need `args.containsKey(DatePickerDialogModule.ARG_SPINNERMODE)`?
`args.getBoolean(DatePickerDialogModule.ARG_SPINNERMODE)` defaults to false

Comment (SilVdB) [inline ReactAndroid/src/main/java/com/facebook/react/modules/datepicker/DatePickerDialogFragment.java]: Indeed, we don't need that condition.

**Comment by janicduplessis** [comment]
Does using https://developer.android.com/reference/android/widget/DatePicker.html#setSpinnersShown(boolean) works instead of using an old holo theme? It would be better if there ends up being a material style spinner date picker.

**Comment by hramos** [comment]
@SilVdB do you still plan to ship this PR?

**Comment by mkonicek** [comment]
Just merged https://github.com/facebook/react-native/pull/10932 which should provide the same functionality.

**Pull request closed without merging**